sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -206,6 +206,17 @@ class SessionCatalog(
     builder.build[QualifiedTableName, LogicalPlan]()
   }
 
+  private val v2TableCache: Cache[QualifiedTableName, Table] = {
+    // Use the same cache size as V1 table cache.
+    var builder = CacheBuilder.newBuilder().maximumSize(cacheSize)
+
+    if (cacheTTL > 0) {
+      builder = builder.expireAfterWrite(cacheTTL, TimeUnit.SECONDS)
+    }
+
+    builder.build[QualifiedTableName, Table]()
```

Comment:
shall we only cache the `TableProvider`? I'm afraid this cache may cause data staleness as the table schema will never change now.